src/Suave/Http.fs

```diff
@@ -788,3 +788,103 @@ module Http =
             }
       }
       |> succeed
+
+  module CORS =
+    
+    open System
+    open Successful
+
+    [<Literal>]
+    let Origin = "Origin"
+    [<Literal>]
+    let AccessControlRequestMethod = "Access-Control-Request-Method"
+    [<Literal>]
+    let AccessControlRequestHeaders = "Access-Control-Request-Headers"
+    [<Literal>]
+    let AccessControlAllowOrigin = "Access-Control-Allow-Origin"
+    [<Literal>]
+    let AccessControlAllowMethods = "Access-Control-Allow-Methods"
+    [<Literal>]
+    let AccessControlAllowHeaders = "Access-Control-Allow-Headers"
+    [<Literal>]
+    let AccessControlAllowCredentials = "Access-Control-Allow-Credentials"
+    [<Literal>]
+    let AccessControlExposeHeaders = "Access-Control-Expose-Headers"
+    [<Literal>]
+    let AccessControlMaxAge = "Access-Control-Max-Age"
+
+    let private isAllowedOrigin config value = 
+      match config.allowedUris with
+      | InclusiveOption.All -> true
+      | InclusiveOption.None -> false
+      | InclusiveOption.Some uris ->
+        List.exists (fun a -> a = value) uris
+
+    let private setMaxAgeHeader config =
+      match config.maxAge with
+      | None -> succeed
+      | Some age -> Writers.setHeader AccessControlMaxAge (age.ToString())
+
+    let private setAllowCredentialsHeader config = 
+      Writers.setHeader AccessControlAllowCredentials (config.allowCookies.ToString())
+
+    let private setAllowMethodsHeader config value =
+      match config.allowedMethods with
+      | InclusiveOption.None -> succeed
+      | InclusiveOption.All -> Writers.setHeader AccessControlAllowMethods "*"
+      | InclusiveOption.Some methods -> 
```

Comment:
You could do `...Some (m :: ms) as methods` and make the below lines less than 80 chars and also easier to read.